ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -1093,6 +1078,7 @@ func (r *RayClusterReconciler) createWorkerPod(ctx context.Context, instance ray
 func (r *RayClusterReconciler) buildHeadPod(ctx context.Context, instance rayv1.RayCluster) corev1.Pod {
 	logger := ctrl.LoggerFrom(ctx)
 	podName := utils.PodGenerateName(instance.Name, rayv1.HeadNode)
+	podName = podName[:len(podName)-1]                                            // strip the last character so the `-` will be excluded from the generated podName
```

Comment:
This is hacky.

1. Update the function name `PodGenerateName` to `PodName(prefix string, nodeType rayv1.RayNodeType, isGenerateName bool)`

2. If `isGenerateName` is true, add a dash at the end. Otherwise, don't add a dash.